src/neo/SmartContract/Native/Tokens/Nep11Token.cs

```
@@ -0,0 +1,225 @@
+
+            var events = new List<ContractEventDescriptor>(Manifest.Abi.Events)
+            {
+                new ContractMethodDescriptor()
+                {
+                    Name = "Transfer",
+                    Parameters = new ContractParameterDefinition[]
+                    {
+                        new ContractParameterDefinition()
+                        {
+                            Name = "from",
+                            Type = ContractParameterType.Hash160
+                        },
+                        new ContractParameterDefinition()
+                        {
+                            Name = "to",
+                            Type = ContractParameterType.Hash160
+                        },
+                        new ContractParameterDefinition()
+                        {
+                            Name = "amount",
+                            Type = ContractParameterType.Integer
+                        },
+                        new ContractParameterDefinition()
+                        {
+                            Name = "tokenId",
+                            Type = ContractParameterType.ByteArray
+                        }
+                    },
+                    ReturnType = ContractParameterType.Boolean
+                }
+            };
+
+            Manifest.Abi.Events = events.ToArray();
+        }
+
+        [ContractMethod(0_01000000, CallFlags.AllowStates)]
+        public virtual BigInteger TotalSupply(StoreView snapshot)
+        {
+            StorageItem storage = snapshot.Storages.TryGet(CreateStorageKey(Prefix_TotalSupply));
+            if (storage is null) return BigInteger.Zero;
+            return new BigInteger(storage.Value);
+        }
+
+        [ContractMethod(0_01000000, CallFlags.AllowStates)]
+        public virtual BigInteger BalanceOf(StoreView snapshot, UInt160 account, byte[] tokenId)
+        {
+            if (tokenId.Length > MaxTokenIdLength) return BigInteger.Zero;
```

`The parameter owner SHOULD be a 20-byte address. If not, this method SHOULD throw an exception.`

We should unify it like https://github.com/neo-project/neo/pull/1670/files#diff-f7ac0fe60b0bcd46ff4d867627d38bf6R112

---

Not necessary, because the incorrectly formatted parameters cannot call the method

---

Sorry, I copied the wrong text, i means that token id should be checked in all methods in the same way, throwing an exception.

---

OK.Got it.